io/disk/scsi_add_remove.py

```diff
@@ -33,44 +36,64 @@ def setUp(self):
+        system_wwids = multipath.get_multipath_wwids()
+        for wwid in self.wwids:
+            if wwid not in system_wwids:
+                self.log.info("%s not present in the system" % wwid)
+
+    def GetscsiId(self, path):
+        '''
+        calculate and return the scsi_id of a disk
+        '''
+        cmd = "lsscsi"
+        out = process.run(cmd)
+        for lines in out.stdout.splitlines():
+            if path in lines:
+                scsi_num = lines.split()[0].strip("[").strip("]")
+                self.log.info("scsi_num=%s", scsi_num)
+                scsi_num = scsi_num.replace(":", " ")
+                return scsi_num
 
     def test(self):
         '''
         Function where test is executed
         '''
-        device_list = []
-        cmd = "ls -l /dev/disk/by-path/"
-        output = process.run(cmd)
-        for lines in output.stdout.splitlines():
-            if self.pci_device in lines:
-                device_list.append(lines.split()[-1])
-        if not device_list:
-            self.log.warning("No devices under the given PCI device")
-        else:
-            for device_id in device_list:
-                device = device_id.split()[-1].strip("../*")
-                self.log.info("device = %s", device)
-                cmd = "lsscsi"
-                output = process.run(cmd)
-                for lines in output.stdout.splitlines():
-                    if device in lines:
-                        scsi_num = lines.split()[0].strip("[").strip("]")
-                        self.log.info("scsi_num=%s", scsi_num)
-                        scsi_num_seperated = scsi_num.replace(":", " ")
-                        self.log.info("Deleting %s", scsi_num)
-                        genio.write_file("/sys/block/%s/device/delete"
-                                         % device, "1")
-                        time.sleep(5)
-                        self.log.info("%s deleted", scsi_num)
-                        self.log.info("adding back %s", scsi_num)
-                        process.run("echo scsi add-single-device %s > \
-                                     /proc/scsi/scsi", scsi_num_seperated)
-                        time.sleep(5)
-                        self.log.info("%s Added back", scsi_num)
+        for wwid in self.wwids:
+            paths = multipath.get_paths(wwid)
+            for path in paths:
+                scsi_id = self.GetscsiId(path)
+                process.run("lsscsi")
+                self.log.info("Deleting %s = %s" % (path, scsi_id))
+                genio.write_file("/sys/block/%s/device/delete" % path, "1")
+                time.sleep(5)
+                new_paths = multipath.get_paths(wwid)
+                if path in new_paths:
+                    self.err_paths.append(path)
+                else:
+                    self.log.info("\n%s = %s deleted\n" % (path, scsi_id))
+                self.log.info("\nadding back %s = %s\n" % (path, scsi_id))
```

Comment:
Consistency in log message format please

Reply:
@narasimhan-v  you mean here to remove the '\n'?

Reply:
Yes. Either have in all the messages, or remove in all.